### ISSUE TITLE
Unable to list images on simplestreams server

### ISSUE THREAD

**Issue by gartnera** [lxc/lxd#6798]
# Issue description

Upon updating to lxd 3.19, I'm unable to list images on my simplestreams server (generated with https://github.com/Avature/lxd-image-server). I wasn't able to obtain any debug logging from `lxc`, operations related to simplestream images don't seem to be logged.

We build images with gitlab ci, you can browse the artifacts [here](https://gitlab.com/container-manager/images/-/jobs/415128159/artifacts/browse/public/).

Output:

```
alex@ubuntu:~$ lxc version
Client version: 3.19
Server version: 3.19
alex@ubuntu:~$ lxc --debug image ls gitlab:
DBUG[01-28|21:47:40] Connecting to a remote simplestreams server
+-------+-------------+--------+-------------+--------------+------+------+-------------+
| ALIAS | FINGERPRINT | PUBLIC | DESCRIPTION | ARCHITECTURE | TYPE | SIZE | UPLOAD DATE |
+-------+-------------+--------+-------------+--------------+------+------+-------------+
alex@ubuntu:~$
```

Expected output (old lxd version):
```
agartner@lxd-test:~$ lxc version
Client version: 3.0.3
Server version: 3.0.3
agartner@lxd-test:~$ lxc --debug image ls gitlab:
DBUG[01-28|21:47:16] Connecting to a remote simplestreams server
+------------------------------+--------------+--------+-------------------------------------+--------+----------+-------------------------------+
|            ALIAS             | FINGERPRINT  | PUBLIC |             DESCRIPTION             |  ARCH  |   SIZE   |          UPLOAD DATE          |
+------------------------------+--------------+--------+-------------------------------------+--------+----------+-------------------------------+
| ubuntu/16.04 (1 more)        | 4108b4de6418 | yes    | ubuntu 16.04 amd64 (20200126_05:06) | x86_64 | 126.97MB | Jan 26, 2020 at 12:00am (UTC) |
+------------------------------+--------------+--------+-------------------------------------+--------+----------+-------------------------------+
| ubuntu/18.04 (1 more)        | 23b9b694b166 | yes    | ubuntu 18.04 amd64 (20200126_05:06) | x86_64 | 179.38MB | Jan 26, 2020 at 12:00am (UTC) |
+------------------------------+--------------+--------+-------------------------------------+--------+----------+-------------------------------+
| ubuntu/18.04/docker (1 more) | 3406984f23c1 | yes    | ubuntu 18.04 amd64 (20200126_05:06) | x86_64 | 199.35MB | Jan 26, 2020 at 12:00am (UTC) |
+------------------------------+--------------+--------+-------------------------------------+--------+----------+-------------------------------+
| ubuntu/19.04 (1 more)        | bad0fe80dcba | yes    | ubuntu 19.04 amd64 (20200126_05:06) | x86_64 | 152.81MB | Jan 26, 2020 at 12:00am (UTC) |
+------------------------------+--------------+--------+-------------------------------------+--------+----------+-------------------------------+
```

I think this worked in 3.18, but I can't seem to find a way to install 3.18 confirm. I looked at the shared/simplestreams commit history a bit, but couldn't immediately identify anything that would cause this. Perhaps there were some protocol changes recently?

# Steps to reproduce

 1. Install lxd 3.19
 2. `lxc remote add --protocol simplestreams gitlab https://container-manager.gitlab.io/images`
 3. `lxc image ls gitlab:`

**Comment by stgraber** [comment]
I'm getting the same result on 3.18.

**Comment by stgraber** [comment]
3.16 to 3.17 is when something changed looks like

**Comment by stgraber** [comment]
Ah, there was a big rework of simplestreams around that time, I wonder if we got stricter about something and it's causing records to now be dropped.

**Comment by stgraber** [comment]
Right, you're effectively relying on a bug in LXD's parser :)

TLDR is, you want to set `combined_squashfs_sha256` when you provide `squashfs` images, not `combined_sha256`.

`combined_sha256` is an old key which predates support for multiple type of rootfs images.
It was used when the only rootfs option was `tar.xz` and is still supported for that purpose when `combined_rootxz_sha256` isn't set.

But that only applies when the rootfs is in fact `rootxz`. In your case, your rootfs is of type `squashfs` so you need to set the `combined_squashfs_sha256` key instead.


As there are many image servers that offer both rootxz and squashfs version of the same image, we really cannot have `combined_sha256` apply to `squashfs` or we're going to get weird hash mismatches (which I suspect is the bug we fixed and which broke parsing of your stream).

**Comment by gartnera** [comment]
Ok I'll try and fix upstream. Thanks for pointing me in the right direction!
